cloud-agent/service/server/src/main/scala/org/hyperledger/identus/pollux/credentialschema/controller/CredentialSchemaControllerImpl.scala

```diff
@@ -30,33 +37,71 @@ class CredentialSchemaControllerImpl(service: CredentialSchemaService, managedDI
       rc: RequestContext
   ): ZIO[WalletAccessContext, ErrorResponse, CredentialSchemaResponse] = {
     for {
-      validated <- validatePrismDID(in.author)
+      _ <- validatePrismDID(in.author)
       result <- service
         .create(toDomain(in))
         .map(cs => fromDomain(cs).withBaseUri(rc.request.uri))
     } yield result
   }
 
-  override def updateSchema(author: String, id: UUID, in: CredentialSchemaInput)(implicit
+  def createSchemaDidUrl(config: AppConfig, in: CredentialSchemaInput)(implicit
       rc: RequestContext
-  ): ZIO[WalletAccessContext, ErrorResponse, CredentialSchemaResponse] = {
-    for {
+  ): ZIO[WalletAccessContext, ErrorResponse, CredentialSchemaDidUrlResponse] = {
+    val res = for {
+      validated <- validatePrismDID(in.author)
+      serviceName = config.agent.httpEndpoint.serviceName
+      result <- service.create(toDomain(in), ResourceResolutionMethod.DID)
+      response <- ZIO
+        .fromEither(CredentialSchemaDidUrlResponse.fromDomain(result, serviceName))
+        .mapError(e =>
+          ErrorResponse.internalServerError(detail = Some(s"Error occurred while parsing a schema response: $e"))
+        )
+
+    } yield response
+
+    res
+  }
+
+  override def updateSchema(config: AppConfig, id: UUID, in: CredentialSchemaInput)(implicit
+      rc: RequestContext
+  ): ZIO[WalletAccessContext, ErrorResponse, CredentialSchemaResponse | CredentialSchemaDidUrlResponse] = {
+    val res = for {
       _ <- validatePrismDID(in.author)
-      result <- service
-        .update(id, toDomain(in).copy(author = author))
-        .map(cs => fromDomain(cs).withBaseUri(rc.request.uri))
+      serviceName = config.agent.httpEndpoint.serviceName
+      cs <- service
+        .update(id, toDomain(in))
+      result <- cs.resolutionMethod match
+        case ResourceResolutionMethod.DID =>
+          ZIO
+            .fromEither(CredentialSchemaDidUrlResponse.fromDomain(cs, serviceName))
+            .mapError(e =>
+              ErrorResponse.internalServerError(detail = Some(s"Error occurred while parsing a schema response: $e"))
+            )
+
+        case ResourceResolutionMethod.HTTP =>
+          ZIO.succeed(fromDomain(cs).withBaseUri(rc.request.uri))
     } yield result
+
+    res
   }
 
-  override def getSchemaByGuid(guid: UUID)(implicit
+  override def getSchemaByGuid(config: AppConfig, guid: UUID)(implicit
```

Comment:
Passing an entire AppConfig object is a code smell.
From a long-term perspective, it creates a coupling between the application conf and the schema API.
Passing the required parameters (or the object with these parameters) to the function would be better.